src/components/ToggleSwitch.js

```diff
@@ -47,23 +56,23 @@ const CheckBox = styled.input`
   }
 `;
 
-const ToggleSwitch = ({ settings, toggleDarkmode }) => {
-  const toggle = settings.darkmode;
+const ToggleSwitch = ({ id, active, onChangeSettings }) => {
+  const [isChecked, setIsChecked] = React.useState(active);
 
-  function handleChange(event) {
-    toggleDarkmode(event.target.checked);
+  function handleChange() {
+    setIsChecked(!isChecked);
+    onChangeSettings(id, isChecked);
```

Comment:
Actually, this should be `onChangeSetting`, because a `ToggleSwitch` only controls one setting.

Reply:
gotcha